reactive/kubernetes_worker.py

```diff
@@ -452,14 +463,28 @@ def watch_for_changes():
         set_state('kubernetes-worker.restart-needed')
 
 
+@when('kubernetes-worker.cloud.pending',
+          'kubernetes-worker.cloud.blocked')
+@when_not('kubernetes-worker.snaps.installed', 'kube-api-endpoint.available',
+      'tls_client.ca.saved', 'tls_client.certs.saved',
+      'kube-control.dns.available', 'kube-control.auth.available',
+      'cni.available', 'kubernetes-worker.restart-needed',
+      'worker.auth.bootstrapped', 'endpoint.container-runtime.available')
```

Comment:
It seems like most of this should actually be in an `@when()` and only the `endpoint.container-runtime.available` in the `@when_not()`.  That, or the only condition be on the container runtime flag.

Reply:
Actually, scratch that.  This should be integrated into the `charm_status` function.